openedx/features/enterprise_support/tests/test_api.py

```diff
@@ -221,6 +289,31 @@ def test_consent_needed_for_course(self, mock_get_enterprise_learner_data):
         self.mock_consent_get(user.username, course_id, ec_uuid)
         self.assertFalse(consent_needed_for_course(request, user, course_id))
 
+        # test when the enrollment already exists without a consent record existing.
+        clear_data_consent_share_cache(user.id, course_id)
+        self.mock_consent_missing(user.username, course_id, ec_uuid)
+        self.assertFalse(consent_needed_for_course(request, user, course_id, enrollment_exists=True))
+
+    @httpretty.activate
+    @mock.patch('openedx.features.enterprise_support.api.get_enterprise_learner_data_from_db')
+    def test_consent_needed_for_course_no_learner_data(self, mock_get_enterprise_learner_data):
+        user = UserFactory(username='janedoe')
+        request = mock.MagicMock(
+            user=user,
+            site=SiteFactory(domain="example.com"),
+            session={},
+            COOKIES={},
+            GET={},
+        )
+        ec_uuid = 'cf246b88-d5f6-4908-a522-fc307e0b0c59'
+        course_id = 'fake-course'
+        mock_get_enterprise_learner_data.return_value = None
+        self.mock_enterprise_learner_api()
+
+        # test not required consent for example non enterprise customer
```

Comment:
nit: "test consent not required for non-enterprise customer" sounds more natural

Reply:
Hah, that looks like some copypasta, I'll try to fix it elsewhere, too.